tests/urls.py

```diff
@@ -0,0 +1,22 @@
+from django.conf.urls import re_path
+from rest_framework.routers import SimpleRouter
+
+from .views import BasicModelRelationshipView, BasicModelViewSet
+
+router = SimpleRouter()
+router.register(r"basic_models", BasicModelViewSet, basename="basic-model")
+
+urlpatterns = [
```

Comment:
I am a bit on the fence with this as we are currently just migrating to the new test module and are not sure how things are turning out. But one issue in the past we had is that we had too much shared test code between different tests making maintanance tricky. As this is currently only needed in test_relations.py I prefer we move it into that module for now. We can always extract it again if need to later on.

Reply:
Your reasoning definitely makes sense. However, I don't know how to override the urlconf machinery besides using the decorator `@pytest.mark.urls()` and pointing to a separate module. Do you know how to tell Django to use a urlconf that is "inline" within the test module? If you can point me in the right direction, I'm happy to make this change.

Reply:
The urlpatterns needs to be moved into `test_relations.py` top level and then you can use the following url marker:

```python
@pytest.mark.urls(__name__)
```

In [test_parsers.py](https://github.com/django-json-api/django-rest-framework-json-api/blob/d8f26fc9f87a8c684a80eafb10fe05524f3b3856/example/tests/test_parsers.py) we already do this but instead of using the urls marker it overwrites [ROOT_URLCONF](https://github.com/django-json-api/django-rest-framework-json-api/blob/d8f26fc9f87a8c684a80eafb10fe05524f3b3856/example/tests/test_parsers.py#L136) setting. I prefer the marker though.

Reply:
Perfect, I'll give that a shot. Thanks!